pikitlib/pikitlib/ultrasonic.py

```diff
@@ -0,0 +1,76 @@
+import RPi.GPIO as GPIO
+import time
+import statistics
+
+#Class requirements (from WPILIB):
+# 
+
+
+class Ultrasonic:
+    def __init__(self):
+        GPIO.setwarnings(False)
+        self.trigger_pin = 27#=pingChannel
+        self.echo_pin = 22#echoChannel  
+        self.chirp_length = 0.00015
+        self.detect_value = 10000 #This is the value at which
+        #we report the signal having been received. This value
+        #is from freenove.
+        GPIO.setmode(GPIO.BCM)
+        GPIO.setup(self.trigger_pin,GPIO.OUT)
+        GPIO.setup(self.echo_pin,GPIO.IN)
+    #Sets up pins connected to the sensor pulse sender 
+    # and receiver. On rio would be replaced with channels.
+
+    def send_trigger_pulse(self):
+        chirp_length = self.chirp_length
+        GPIO.output(self.trigger_pin,True)
+        time.sleep(chirp_length) 
+        '''
+        This is the duration of the 
+        chirp (will be a little longer because its sound)
+        We may want to decrease the duration; the wpilib
+        Ultrasonic suggests 10e^-6 = 0.00001s as a chirp,
+        but this is for the
+        '''
+        GPIO.output(self.trigger_pin,False)
+
+    def wait_for_echo(self,value,timeout):
```

Comment:
Please document the units for the timeout.     This is effectively the range of the sonar, but it should be clear where this number comes from.   In fact, it might be better if the timeout was derived from the range?